### DISCUSSION TITLE
problem about dark-mode

### DISCUSSION THREAD

**Discussion by tsq456** [tailwindlabs/tailwindcss#5156] [Help] [answered]
I found that after I turn on the dark mode, sometimes when I switch the page, the page style first changes to white and then quickly changes to black.

Is this caused by TailwindCSS or my code’s problem?

My site's demo here, https://610beea733a13bf0419dac88--determined-noether-276811.netlify.app/, and you can find the switcher in the top-left.

git repository here, https://github.com/tsq456/hugo-theme-windy

Looking forward to the answer!

**Comment by dance2die** [comment, marked as answer]
Looks like you'd have to block page rendering to fetch theme data (https://stackoverflow.com/a/63033934) from localstorage before loading the page.

![dark theme flickers](https://user-images.githubusercontent.com/8465237/128396807-765ff6a5-0020-4088-afea-17939e8e3b06.png)

The profiler shows the flicker during page load because the script runs later to apply the dark theme (conditionally retrieving the theme data from localstorage).

**Reply by tsq456** [reply]
Thanks a lot, it really helped me. I have solved the problem!

thanks again.
